### ISSUE TITLE
Alert enhancement

### ISSUE THREAD

**Issue by cheton** [trendmicro-frontend/tonic-ui#110]
https://material-ui.com/components/alert/

* ~Add `AlertTitle` component to display a formatted title above the content.~
* Rename `status` to `severity` to reflect `success`, `info`, `warning`, `error`
* (optional) Add `variant` prop: `filled` and `outline`

**Comment by cheton** [comment]
@austinhung81 @joshuatai @roth1002 @tinaClin 

I would like to refactor the `Alert` component as below to simplify its usage. What are about your thoughts?

```jsx
<Alert severity="success">
  This is a success alert.
</Alert>

<Alert severity="info">
  This is an info alert.
</Alert>

<Alert severity="success" icon={false}>
  This is a success alert without icon.
</Alert>

<Alert severity="success" icon="my-icon">
  This is a success alert with a different icon.
</Alert>

<Alert severity="success" icon={<Icon name="my-icon" color="gray:50" />}>
  This is a success alert with a custom icon component.
</Alert>

<Alert severity="success">
  <AlertTitle>Success</AlertTitle>
  This is a success alert with a formatted title above the content.
</Alert>
```

**Comment by austinhung81** [comment]
- Although `AlertTitle` is not usually used in our product, but that will be a good feature for the customization.
- Revise `status` is also more clearly. 
- `Variant` props can arrange a discussion with VDs.

**Comment by cheton** [comment]
The `AlertTitle` component might not be frequently used, instead of introducing a new component, I will add an example to demonstrate how to implement this feature when necessary.

```jsx
const AlertTitle = (props) => (
  <Box mt={-2} fontWeight="semibold" {...props} />
);
```

**Comment by cheton** [comment]
Fixed in PR #112
